InterneTaakAfhandeling.Web.Server/Features/MyInterneTakenOverview/MyInterneTakenOverviewService.cs

```diff
@@ -1,86 +1,97 @@
-﻿
-using InterneTaakAfhandeling.Common.Services.OpenKlantApi;
+﻿using InterneTaakAfhandeling.Common.Services.OpenKlantApi;
 using InterneTaakAfhandeling.Common.Services.OpenKlantApi.Models;
 using InterneTaakAfhandeling.Web.Server.Authentication;
+using InterneTaakAfhandeling.Web.Server.Features.Internetaken;
 
-namespace InterneTaakAfhandeling.Web.Server.Services
+namespace InterneTaakAfhandeling.Web.Server.Features.MyInterneTakenOverview
 {
     public interface IUserService
     {
-        Task<IReadOnlyList<Internetaak>> GetInterneTakenByAssignedUser(ITAUser user);
+        Task<IReadOnlyList<Internetaak>> GetInterneTakenByAssignedUser(ITAUser user, bool? afgerond);
     }
-    public class UserService(IOpenKlantApiClient openKlantApiClient) : IUserService
+    public class MyInterneTakenOverviewService(IOpenKlantApiClient openKlantApiClient) : IUserService
     {
-        private readonly IOpenKlantApiClient _openKlantApiClient = openKlantApiClient;
-
-
-        public async Task<IReadOnlyList<Internetaak>> GetInterneTakenByAssignedUser(ITAUser user)
-        {
-            var actorIds = await GetActorIds(user);
-            var internetakenTasks = actorIds.Select(a => _openKlantApiClient.GetOutstandingInternetakenByToegewezenAanActor(a));
-
-            var results = await Task.WhenAll(internetakenTasks);
-
+        private readonly IOpenKlantApiClient _openKlantApiClient = openKlantApiClient;
+
+
+        public async Task<IReadOnlyList<Internetaak>> GetInterneTakenByAssignedUser(ITAUser user, bool? afgerond)
+        {
+            var actorIds = await GetActorIds(user);
+
+            var internetakenTasks = actorIds
+                .Where(id => Guid.TryParse(id, out _))
+                .Select(actorId =>
+                {
+                    var query = new InterneTaakQuery
+                    {
+                        ToegewezenAanActor__Uuid = Guid.Parse(actorId)
+                    };
+                    query.Status = (afgerond == true)  ? KnownInternetaakStatussen.Verwerkt : KnownInternetaakStatussen.TeVerwerken;
```

Comment:
With an optional boolean query parameter, I would expect:
```suggestion
                    query.Status = afgerond switch
{
    true => KnownInternetaakStatussen.Verwerkt,
    false => KnownInternetaakStatussen.TeVerwerken,,
    _ => null
};
```

Reply:
ik heb het toch anders opgelost. standaard zou je anders alle contactverzoeken krijgen en het lijkt me ondenkbaar dat daar een use case voor zou kunnen zijn, dus ik heb de parameter niet-optioneel gemaakt